src/library/itunes/itunesmacosimporter.mm

```diff
@@ -96,9 +96,12 @@ bool isPlaylistShown(ITLibPlaylist* playlist) {
         // Filter out the primary playlist (since we already show the library
         // under the main iTunes node)
         bool isPrimary;
+#if (MAC_OS_X_VERSION_MAX_ALLOWED >= 120000)
         if (@available(macOS 12.0, *)) {
             isPrimary = [playlist isPrimary];
-        } else {
+        } else
+#endif
```

Comment:
whats the error here? I figured if `@available` is asking for a version lower than the current SDK it would just return `false`. the official docs don't specify that unfortunately. https://developer.apple.com/documentation/swift/marking-api-availability-in-objective-c

Reply:
The error was "isPrimary() is not a member of ..." 

This is because you need the maximum SDK at build time to have all used definitions in the headers. 

Your assumption is correct for the runtime check of `@available`

Reply:
Ah that makes sense. Thanks for the explanation. Isn't the `@available` check redundant in that case?

Reply:
You need both in case of a new SDK build executed on an old macOS. Than the `#if` is true and `@available` false.

Reply:
Ah right, so one check is compile-time one is runtime...